src/common/thread_queue_list.h

```diff
@@ -157,6 +160,52 @@ struct ThreadQueueList {
     Queue* first;
     // The priority level queues of thread ids.
     std::array<Queue, NUM_QUEUES> queues;
+
+    s32 ToIndex(Queue* q) const {
+        if (q == nullptr) {
+            return -2;
+        } else if (q == UnlinkedTag()) {
+            return -1;
+        } else {
+            return static_cast<s32>(q - &queues[0]);
+        }
+    }
+
+    Queue* ToPointer(s32 idx) {
+        if (idx == -1) {
+            return UnlinkedTag();
+        } else if (idx < 0) {
+            return nullptr;
+        } else {
+            return &queues[idx];
+        }
+    }
+
+    friend class boost::serialization::access;
+    template <class Archive>
+    void save(Archive& ar, const unsigned int file_version) const {
+        s32 idx = ToIndex(first);
+        ar << idx;
+        for (auto i = 0; i < NUM_QUEUES; i++) {
```

Comment:
```suggestion
        for (size_t i = 0; i < NUM_QUEUES; i++) {
```

Otherwise this will cause a sign mismatch warning, since `auto i = 0;` forms an int.